src/pages/groupComparison/GroupComparisonMutationMapper.tsx

```diff
@@ -214,4 +357,155 @@ export default class GroupComparisonMutationMapper extends MutationMapper<
+                values: [value],
+            },
+        }));
+
+        let groupedData = groupDataByGroupFilters(
+            filters,
+            allGroupedData[groupIndex].data,
+            this.store.dataStore.applyFilter
+        );
+
+        return _.keyBy(groupedData, d => d.group);
+    }
+
+    @autobind
+    protected mutatedCountsByProteinChangeForGroup(
+        groupIndex: number
+    ): {
+        [proteinChange: string]: number;
+    } {
+        const map: { [proteinChange: string]: number } = {};
+
+        _.forIn(
+            this.mutationsGroupedByProteinChangeForGroup(groupIndex),
+            (v, k) => {
+                const uniqueMutations = _.uniqBy(v.data, d => d[0].patientId);
+                map[v.group] = uniqueMutations.length;
+            }
+        );
+
+        return map;
+    }
+
+    protected get fisherExactTwoSidedTestLabel(): JSX.Element | null {
+        return (
+            <FisherExactTwoSidedTestLabel
+                dataStore={
+                    this.props.store.dataStore as MutationMapperDataStore
+                }
+                groups={this.props.groups}
+                groupToProfiledPatientCounts={
+                    this.props.groupToProfiledPatientCounts
+                }
+            />
+        );
+    }
+
+    @computed get tableCustomControls(): JSX.Element {
+        return (
+            <div
+                style={{
+                    display: 'flex',
+                    flexDirection: 'row',
+                    alignItems: 'center',
+                    height: 30,
+                    float: 'right',
+                }}
+            >
+                <div style={{ width: 250, marginRight: 7 }}>
+                    <CheckedSelect
+                        name={'groupsSelector'}
+                        placeholder={'Enriched in ...'}
+                        onChange={this.onChange}
+                        options={this.options}
+                        value={this.selectedValues}
+                    />
+                </div>
+                <label className="checkbox-inline" style={{ marginRight: 7 }}>
+                    <input
+                        type="checkbox"
+                        checked={this.significanceFilter}
+                        onClick={this.toggleSignificanceFilter}
+                        data-test="SwapAxes"
+                    />
+                    Significant only
+                </label>
+            </div>
+        );
+    }
+
+    @computed get options(): Option[] {
+        return _.map(this.props.groups, group => {
+            return {
+                label: group.nameWithOrdinal,
+                value: group.nameWithOrdinal,
+            };
+        });
+    }
+
+    @computed get selectedValues() {
+        return this._enrichedGroups.map(id => ({ value: id }));
+    }
+
+    @action.bound
+    onChange(values: { value: string }[]) {
+        this._enrichedGroups = _.map(values, datum => datum.value);
```

Comment:
seems odd that we are doing two things in this handler.  updating enrichedGroups and also whatever onFilterOptionSelect does.  Like, shouldn't the former be included in the latter?